modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -58,6 +67,53 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint>> extends Psbt {
     return psbt;
   }
 
+  static async fromTransactionComplete(
+    transaction: UtxoTransaction<bigint>,
+    prevOutputs: TxOutput<bigint>[],
+    fetchTransactions: (txids: string[]) => Promise<Record<string, Buffer>>
+  ): Promise<UtxoPsbt<UtxoTransaction<bigint>>> {
+    const psbt = this.fromTransaction(transaction, prevOutputs);
+
+    const txidToIndex: Record<string, number> = {};
+    psbt.data.inputs.forEach((input, index) => {
+      if (!input.tapLeafScript && !input.witnessScript) {
+        txidToIndex[reverseBuffer(transaction.ins[index].hash).toString('hex')] = index;
+      }
+    });
```

Comment:
it appears that this will not handle multiple inputs from the same transaction correctly (only the last one will be considered)